spec/Appendix B -- Grammar Summary.md

```diff
@@ -258,6 +260,8 @@ TypeSystemExtension :
 - SchemaExtension
 - TypeExtension
 
+FullSchemaDocument : TypeSystemDefinition+
```

Comment:
Interesting that we are disallowing extensions in the FullSchemaDocument. I'm not opposed but this is interesting.

That feels a little odd, as the Relay Compiler tends to overload extension to mean client-defined (this is probably wrong and we should switch to using a directive).

Reply:
I went with `no extension` by default to make a full schema as simple and as easy to read as possible. 

A "full schema" is probably machine written and probably machine consumed as well. It's the source of truth for a service. 

Another typical use case besides validation would be someone clicking their way in IntelliJ to find a type definition. I'd argue that having the extensions merged at that point would be nicer because you don't have to remember to "merge" things.

That being said, I think it'd be OK to "extend" a full schema:

`client full schema` = `server full schema` + `type extensions`

We'll probably end up encouraging this in Apollo Kotlin, possibly for stuff like `@nullOnlyOnError`